### PR TITLE
Google Sign-in sample: Use GoogleSignInClient

### PR DESCRIPTION
**Pull request by jush**
This PR uses the GoogleSignInClient instead of GoogleApiClient.
Also, use the already available com.google.android.gms.common.SignInButton instead of custom one.
Updated target and compile SDK version to 29.

Moreover, some dependencies were not needed.

### CODE REVIEW

File: WearStandaloneGoogleSignIn/app/build.gradle [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-
-    buildToolsVersion "28.0.3"
+    compileSdkVersion 29
 
     defaultConfig {
         applicationId "com.example.android.wearable.wear.wearstandalonegooglesignin"
 
         minSdkVersion 25
-        targetSdkVersion 28
+        targetSdkVersion 29
```
Comment (codingjeremy) [inline WearStandaloneGoogleSignIn/app/build.gradle]: Still want to target 28 as Wear OS hasn't been released yet with Q (29).

File: WearStandaloneGoogleSignIn/app/build.gradle [hunk 2]
```diff
@@ -23,13 +21,8 @@ android {
 }
 
 dependencies {
-    implementation 'com.google.android.gms:play-services-wearable:17.0.0'
```
Comment (codingjeremy) [inline WearStandaloneGoogleSignIn/app/build.gradle]: I'm guessing we aren't using any capabilityApis, DataApis, ChannelApis, or NodeApis in here. You can ignore this comment unless I add a followup saying I found one. Thanks!

Comment (codingjeremy) [reply WearStandaloneGoogleSignIn/app/build.gradle]: Ok, we don't, you can ignore.

File: WearStandaloneGoogleSignIn/app/build.gradle (hunk 1 above)
Comment (jush) [reply WearStandaloneGoogleSignIn/app/build.gradle]: Fixed in 5c2e340

**Comment by jush** [comment]
Please notice that I've removed the `mUserIdToken` since we are not requesting it as part of the sign in options.

**Comment by jush** [comment]
Btw, would you like me to also update to version 2.5.0 of wearable libraries?

**Comment by codingjeremy** [comment]
Sounds good to me on mUserIdToken. Rest looks good.

Yeah, might as well update this sample to 2.5.0. I can merge pull request after you do that.

Thanks again!!! :)

**Comment by jush** [comment]
No problem. Libraries updated

**Pull request merged**